### PR TITLE
Add maps

### PR DESCRIPTION
**Pull request by zJeremiah**

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -417,6 +419,28 @@ func BuildSchema(title, desc string, example bool, body any) (s Schema, err erro
...
 	switch kind {
+	case reflect.Map:
+		s.Type = Object.String()
+		s.AddProperties = &Schema{}
+		// for now, only getting the type for the first map value
+		// if the type is an interface it will only represent the first value of that interface
+		key := value.MapKeys()[0]
+		val := value.MapIndex(key)
+
+		simple, k, f := isSimpleType(val.Kind())
```
Comment (jbsmith7741) [inline build.go:430]: This function name is unclear. We are asking a yes/no question, but returning 3 values. This probably needs refactoring ... Maybe return a struct with the values that you need and a method to check if it is a basic type or not 
``` go 
type myType struct {
  Kind string 
  Format string 
}
```

File: build_test.go [hunk 2]
```diff
@@ -11,6 +11,11 @@ import (
 
 func TestBuildSchema(t *testing.T) {
 
+	type Simple struct {
+		F1 int    `json:"field_one"`
+		F2 string `json:"field_two"`
+	}
+
```
Comment (jbsmith7741) [inline build_test.go:18]: Is this used somewhere? Its unclear of its purpose.

File: build.go (hunk 1 above)
Comment (jbsmith7741) [reply build.go:430]: This should be called primitive types are that seems to be what your are filtering out.

File: build.go [hunk 3]
```diff
@@ -157,6 +158,7 @@ type Route struct {
 	Requests  map[string]RouteReq   // key reference for requests
 }
 
+// RouteResp is a simplified definition for a Response to a given method
```
Comment (jbsmith7741) [inline build.go:161]: Simplified definition of what Response? http.Response?

File: build.go [hunk 4]
```diff
@@ -417,6 +419,28 @@ func BuildSchema(title, desc string, example bool, body any) (s Schema, err erro
...
+		key := value.MapKeys()[0]
+		val := value.MapIndex(key)
+
+		simple, k, f := isSimpleType(val.Kind())
+		if simple {
+			s.AddProperties = &Schema{
+				Type:   k,
+				Format: f,
+			}
+		} else {
```
Comment (jbsmith7741) [inline build.go:436]: Instead of a nested else statement use a return to reduce indents.

File: build_test.go [hunk 5]
```diff
@@ -57,6 +62,10 @@ func TestBuildSchema(t *testing.T) {
 		print bool // print out the json value
 	}
 
+	type MapTest struct {
+		F1 map[string]Simple `json:"map_field"`
+	}
+
```
Comment (jbsmith7741) [inline build_test.go:68]: Is this used somewhere?

File: build.go (hunk 1 above)
Comment (zJeremiah) [reply build.go:430]: I've updated this to a struct

File: build_test.go (hunk 2 above)
Comment (zJeremiah) [reply build_test.go:18]: this was just to mean a struct with simple types to test

File: build.go (hunk 3 above)
Comment (zJeremiah) [reply build.go:161]: the openapi response object is overly complicated, this object is used to try and simplify the logic and types needed for a response within this code base.

File: build_test.go (hunk 5 above)
Comment (zJeremiah) [reply build_test.go:68]: nope, I'll remove it

**Pull request merged**